plugins/slack_app/app/controllers/slack_app_controller.rb

```diff
@@ -41,32 +42,46 @@ def interact
 
   private
 
+  # TODO
+  # - returning parts of the text can most likely be abused to do some kind of reflection attack.
+  # - / is a bad separator between project and branch since branches often include / too
   def command_response
+    return unknown_user_response unless @current_user_from_slack
+
     # Interesting parts of params:
     # {
     #   "user_id"=>"U0HAGH3AB", <-- look up Samson user using this
     #   "command"=>"/deploy",
-    #   "text"=>"foo bar baz", <-- this is what to deploy
+    #   "text"=>"foobar/master to production", <-- deploy project foobar, branch master to stage production
     #   "response_url"=>"https://hooks.slack.com/commands/T0HAGP0J2/58604540277/g0xd4K2KOsgL9zXwR4kEc0eL"
     #    ^^^^ This is how to respond later, we can also directly return some JSON
     # }
-    return unknown_user unless @current_user_from_slack
-
-    # Parse the command
-    project_name, ref_name, stage_name = @payload['text'].match(/([^\s\/]+)(?:\/(\S+))?\s*(?:to\s+(.*))?/).captures
+    project_permalink, ref_name, stage_permalink =
+      @payload['text'].match(/^([^\s\/]+)(?:\/(\S+))?(?: to (\S+))?$/)&.captures
```

Comment:
Instead of this funky regexp, why not simply split on a space character and use the first 3 results? That way, any git-strategy (including git-flow) with multiple slashes can be supported as well.

Reply:
- this supports multiple slashes
 - I'm trying to keep the api the same, if we want to change the api, then that's another PR